src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,166 @@
+      ErrorCheck.errREV(m_Arm.setIdleMode(IdleMode.kBrake)); //sets brakes when there is no motion
+      
+      ErrorCheck.errREV(m_Arm.setSoftLimit(SoftLimitDirection.kForward, Config.ArmConfig.arm_forward_limit));
+      ErrorCheck.errREV(m_Arm.setSoftLimit(SoftLimitDirection.kReverse, Config.ArmConfig.arm_reverse_limit));
+      ErrorCheck.errREV(m_Arm.enableSoftLimit(SoftLimitDirection.kForward, Config.ArmConfig.SOFT_LIMIT_ENABLE));
+      ErrorCheck.errREV(m_Arm.enableSoftLimit(SoftLimitDirection.kReverse, Config.ArmConfig.SOFT_LIMIT_ENABLE));
+
+      m_Arm.setPeriodicFramePeriod(PeriodicFrame.kStatus5, 20);
+      m_Arm.setPeriodicFramePeriod(PeriodicFrame.kStatus6, 20);
+
+      m_AbsEncoder = m_Arm.getAbsoluteEncoder(Type.kDutyCycle);
+      m_AbsEncoder.setInverted(true);
+      m_AbsEncoder.setPositionConversionFactor(Config.ArmConfig.armPositionConversionFactor);
+      m_AbsEncoder.setVelocityConversionFactor(Config.ArmConfig.armVelocityConversionFactor);
+      m_AbsEncoder.setZeroOffset(Math.toRadians(Config.ArmConfig.armAbsEncoderOffset));
+
+      m_pidControllerArm = m_Arm.getPIDController();
+      m_pidControllerArm.setFeedbackDevice(m_AbsEncoder);
+
+      
+    NetworkTable ArmTuningTable = NetworkTableInstance.getDefault().getTable(m_tuningTable);
+    m_ArmPSubs = ArmTuningTable.getDoubleTopic("P").getEntry(Config.ArmConfig.arm_kP, PubSubOption.periodic(0.02));
+    m_ArmISubs = ArmTuningTable.getDoubleTopic("I").getEntry(Config.ArmConfig.arm_kI);
+    m_ArmDSubs = ArmTuningTable.getDoubleTopic("D").getEntry(Config.ArmConfig.arm_kD);
+    m_ArmIzSubs = ArmTuningTable.getDoubleTopic("IZone").getEntry(Config.ArmConfig.arm_kIz);
+    m_ArmFFSubs = ArmTuningTable.getDoubleTopic("FF").getEntry(Config.ArmConfig.arm_kFF);
+   //m_topArmOffset = topArmTuningTable.getDoubleTopic("Offset").getEntry(ArmConfig.top_arm_offset);
+    momentToVoltageConversion = ArmTuningTable.getDoubleTopic("VoltageConversion").subscribe(m_VoltageConversion);
+
+    m_ArmFFSubs.setDefault(Config.ArmConfig.arm_kFF);
+    m_ArmPSubs.setDefault(Config.ArmConfig.arm_kP);
+    m_ArmISubs.setDefault(Config.ArmConfig.arm_kI);
+    m_ArmDSubs.setDefault(Config.ArmConfig.arm_kD);
+    m_ArmIzSubs.setDefault(Config.ArmConfig.arm_kIz);
+
+    NetworkTable ArmDataTable = NetworkTableInstance.getDefault().getTable(m_dataTable);
+
+    
+    m_ArmPosPub = ArmDataTable.getDoubleTopic("MeasuredAngleDeg").publish(PubSubOption.periodic(0.02));
+    m_TargetAngle = ArmDataTable.getDoubleTopic("TargetAngle").publish(PubSubOption.periodic(0.02));
+
+    ErrorCheck.errREV(m_pidControllerArm.setP(m_ArmPSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setI(m_ArmISubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setD(m_ArmDSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setIZone(m_ArmIzSubs.get())); 
+    ErrorCheck.errREV(m_pidControllerArm.setOutputRange(Config.ArmConfig.min_output, Config.ArmConfig.max_output));
```

Comment:
Can remove the above few lines of code that setP and setI, etc. Anything that is a duplicate with the code in updatePIDSettings